File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>mp;int sum=0;int count=0;
        mp[0]=1;
        for( auto num: nums){
            sum+=num;
            if(mp.find( sum-k)!=mp.end()){
            count+=mp[sum-k];}
            mp[sum]++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int currsum = 0, count = 0;
        Map<Integer, Integer> map = new HashMap<>();

        map.put(0, 1);

        for (int num : nums) {
            currsum += num;
            if (map.containsKey(currsum - k)) {
                count += map.get(currsum - k);
            }
            map.put(currsum, map.getOrDefault(currsum, 0) + 1);
        }

        return count;
    }
}